patches/net/minecraft/world/level/block/entity/BlockEntity.java.patch

```diff
@@ -18,6 +18,15 @@
  
      public BlockEntity(BlockEntityType<?> p_155228_, BlockPos p_155229_, BlockState p_155230_) {
          this.type = p_155228_;
+@@ -51,7 +_,7 @@
+     }
+ 
+     public boolean isValidBlockState(BlockState p_353131_) {
+-        return this.type.isValid(p_353131_);
```

Comment:
Should we additionally deprecate usage of `this.type`?

Reply:
It's private, and nobody should have a reason to AT it when `getType` exists, so I don't see the advantage.

Reply:
Just to be extra clear? Who knows what people might do. :smile: We similarly deprecate some private fields when we use the field redirector.